internal/k6runner/local.go

```diff
@@ -180,7 +180,14 @@ func (r Local) Run(ctx context.Context, script Script, secretStore SecretStore)
 		fmt.Fprintf(logs, `level=error msg="Metrics output truncated at %d bytes"`+"\n", maxMetricsSizeBytes)
 	}
 
-	return &RunResponse{Metrics: metrics.Bytes(), Logs: logs.Bytes()}, errors.Join(err, errorFromLogs(logs.Bytes()))
+	rr := &RunResponse{Metrics: metrics.Bytes(), Logs: logs.Bytes()}
+	if err := errorFromLogs(logs.Bytes()); err != nil {
+		// k6 was run currectly, but the script failed. We capture that failure in the RunResponse.
+		rr.ErrorCode = errorType(err)
+		rr.Error = err.Error()
+	}
+
+	return rr, nil
```

Comment:
This is the highlight of this PR: Instead of returning `errorFromLogs`, it is handled and its result gets put into `RunResponse`. Everything else is instrumental.